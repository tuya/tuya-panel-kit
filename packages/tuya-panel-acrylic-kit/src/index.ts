export { AcrylicBlockCard } from 'tuya-panel-style-block-card';
export { AcrylicButton } from 'tuya-panel-style-button';
export { AcrylicDepictCard, AcrylicDepictIconCard } from 'tuya-panel-style-depict-card';
export { AcrylicDisplayCard } from 'tuya-panel-style-display-card';
export { DataItem, AcrylicEnumButtonGroup } from 'tuya-panel-style-enum-button-group';
export { AcrylicEnumCard } from 'tuya-panel-style-enum-card';
export { AcrylicEnumSliderCard } from 'tuya-panel-style-enum-slider-card';
export { AcrylicIconBlockCard } from 'tuya-panel-style-icon-card';
export { AcrylicSliderCard } from 'tuya-panel-style-slider-card';
export { AcrylicSwitchCard, AcrylicItemCard, AcrylicArrowCard } from 'tuya-panel-style-switch-card';
