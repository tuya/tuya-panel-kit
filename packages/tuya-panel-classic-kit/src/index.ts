export { ClassicBlockCard } from 'tuya-panel-style-block-card';
export { ClassicButton } from 'tuya-panel-style-button';
export { ClassicButtonCard } from 'tuya-panel-style-button-card';
export { ClassicDepictCard } from 'tuya-panel-style-depict-card';
export { ClassicDisplayCard } from 'tuya-panel-style-display-card';
export { ClassicEnumCard } from 'tuya-panel-style-enum-card';
export { ClassicEnumTabsButtonCard } from 'tuya-panel-style-enum-tabs-button-card';
export { ClassicIconBackground } from 'tuya-panel-style-icon-background';
export { ClassicIconBlockCard } from 'tuya-panel-style-icon-card';
export { ClassicSliderCard, ClassicLargeSliderCard } from 'tuya-panel-style-slider-card';
export { ClassicStepCard } from 'tuya-panel-style-step-card';
export { ClassicSwitchCard } from 'tuya-panel-style-switch-card';
