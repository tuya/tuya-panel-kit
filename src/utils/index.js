import ColorUtils from './color';
import CoreUtils from './core';
import JsonUtils from './json';
import NumberUtils from './number';
import RatioUtils from './ratio';
import StringUtils from './string';
import TemperatureUtils from './temperature';
import TimeUtils from './time';
import ThemeUtils from './theme';

export {
  ColorUtils,
  CoreUtils,
  JsonUtils,
  NumberUtils,
  RatioUtils,
  StringUtils,
  TemperatureUtils,
  TimeUtils,
  ThemeUtils,
};
