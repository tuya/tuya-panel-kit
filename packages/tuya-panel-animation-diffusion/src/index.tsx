import React, { Component } from 'react';
import { View, Animated } from 'react-native';
import { Utils } from 'tuya-panel-utils';
import { createAnimation, DIFFUSION_DEFAULT_ANIMATION_CONFIG } from './utils';
import { DiffusionProps } from './interface';

const { convertX: cx } = Utils.RatioUtils;

export default class Diffusion extends Component<DiffusionProps> {
  static defaultProps: DiffusionProps = {
    radius: cx(50),
    maxRadius: cx(100),
    color: '#00FF00',
    width: cx(5),
    number: 2,
    intervalTime: 0,
    startAnimated: true,
    mainDelay: 1000,
    animationConfig: DIFFUSION_DEFAULT_ANIMATION_CONFIG,
    style: {},
    children: null,
  };

  constructor(props) {
    super(props);
    const obj = {};
    for (let i = 0; i < this.props.number; i++) {
      obj[`stateRadius${i}`] = new Animated.Value(props.radius);
      obj[`stateOpacity${i}`] = new Animated.Value(0);
    }
    this.state = obj;
  }

  componentDidMount = () => {
    this.props.startAnimated && this.startAnimated();
  };

  componentWillReceiveProps = nextProps => {
    if (nextProps.startAnimated !== this.props.startAnimated) {
      if (nextProps.startAnimated) {
        this.startAnimated();
      } else {
        this.stopAnimated();
      }
    }
  };

  componentWillUnmount = () => {
    this.stopAnimated();
  };

  timeHandle = [];
  timeHandle1 = [];
  stopAnimated = () => {
    const obj = {};
    for (let i = 0; i < this.props.number; i++) {
      this.state[`stateRadius${i}`].stopAnimation();
      this.state[`stateOpacity${i}`].stopAnimation();
      obj[`stateOpacity${i}`] = new Animated.Value(0);
      obj[`stateRadius${i}`] = new Animated.Value(this.props.radius);
    }
    this.setState(obj);
    this.timeHandle.map(time => clearTimeout(time));
    this.timeHandle = [];
    this.timeHandle1.map(time => clearTimeout(time));
    this.timeHandle1 = [];
  };

  startAnimated = () => {
    for (let i = 0; i < this.props.number; i++) {
      this.childTask(i, true);
    }
  };

  childTask = (i, isInit = false) => {
    const animationConfig = {
      ...DIFFUSION_DEFAULT_ANIMATION_CONFIG,
      ...this.props.animationConfig,
    };
    const { duration, delay, easing, useNativeDriver, isInteraction } = animationConfig;
    this.timeHandle.push(
      setTimeout(
        () => {
          this.setState(
            {
              [`stateOpacity${i}`]: new Animated.Value(1),
              [`stateRadius${i}`]: new Animated.Value(this.props.radius),
            },
            () => {
              Animated.parallel([
                createAnimation({
                  value: this.state[`stateRadius${i}`],
                  toValue: this.props.maxRadius,
                  duration,
                  delay,
                  easing,
                  useNativeDriver,
                  isInteraction,
                }),
                createAnimation({
                  value: this.state[`stateOpacity${i}`],
                  toValue: 0,
                  duration,
                  delay,
                  easing,
                  useNativeDriver,
                  isInteraction,
                }),
              ]).start(() => {
                this.timeHandle1.push(
                  setTimeout(() => {
                    this.childTask(i);
                  }, this.props.intervalTime)
                );
              });
            }
          );
        },
        isInit ? i * this.props.mainDelay : 0
      )
    );
  };

  render() {
    const { width, color, radius, maxRadius, style, number, children } = this.props;
    return (
      <View
        style={[
          {
            justifyContent: 'center',
            alignItems: 'center',
            width: 2 * maxRadius,
            height: 2 * maxRadius,
          },
          style,
        ]}
      >
        {new Array(number)
          .fill(0)
          .map((__, i) => i)
          .map((item, index) => {
            const innerRadius = this.state[`stateRadius${index}`].interpolate({
              inputRange: [radius, maxRadius],
              outputRange: [radius, maxRadius],
            });
            const innerWidth = this.state[`stateRadius${index}`].interpolate({
              inputRange: [radius, maxRadius],
              outputRange: [2 * radius, 2 * maxRadius],
            });
            return (
              <Animated.View
                key={item}
                style={[
                  { position: 'absolute', borderWidth: width, borderColor: color },
                  {
                    width: innerWidth,
                    height: innerWidth,
                    borderRadius: innerRadius,
                    opacity: this.state[`stateOpacity${index}`],
                  },
                ]}
              />
            );
          })}
        <View
          style={{
            width: 2 * radius,
            height: 2 * radius,
            justifyContent: 'center',
            alignItems: 'center',
          }}
        >
          {children}
        </View>
      </View>
    );
  }
}
