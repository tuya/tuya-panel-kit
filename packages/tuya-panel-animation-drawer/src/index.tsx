import _ from 'lodash';
import React, { Component } from 'react';
import { View, Animated, PanResponder, GestureResponderEvent } from 'react-native';
import { Utils } from 'tuya-panel-utils';
import { createAnimation, DRAWER_DEFAULT_ANIMATION_CONFIG } from './utils';
import { DrawerPropTypes, DrawerStateTypes } from './interface';

const { winWidth, winHeight } = Utils.RatioUtils;

export default class Drawer extends Component<DrawerPropTypes, DrawerStateTypes> {
  static defaultProps: DrawerPropTypes = {
    visible: false,
    maskStyle: {
      width: winWidth,
      height: winHeight,
      backgroundColor: 'rgba(0,0,0,0.6)',
      position: 'absolute',
    },
    drawerStyle: {
      backgroundColor: '#F8F8F8',
    },
    style: {},
    placement: 'left',
    hasMask: true,
    maskClosable: true,
    children: null,
    onMaskPress: () => null,
    onStateChange: () => null,
    width: winWidth / 2,
    height: winHeight,
    animationConfig: DRAWER_DEFAULT_ANIMATION_CONFIG,
  };

  constructor(props: DrawerPropTypes) {
    super(props);
    const { placement, visible } = props;
    this.direction = ['left', 'right'].includes(placement) ? 'row' : 'column';
    this.range = ['left', 'right'].includes(placement) ? winWidth : winHeight;
    this.endOnce = true;
    this.state = {
      boxLeft: new Animated.Value(visible ? 0 : -this.range),
      maskOpacity: new Animated.Value(+visible),
      maskState: visible,
    };
  }

  componentWillReceiveProps = (nextProps: DrawerPropTypes) => {
    const { placement, visible, onStateChange } = nextProps;
    if (placement !== this.props.placement) {
      this.direction = ['left', 'right'].includes(placement) ? 'row' : 'column';
      this.range = ['left', 'right'].includes(placement) ? winWidth : winHeight;
    }
    if (visible !== this.props.visible) {
      const animationConfig = {
        ...DRAWER_DEFAULT_ANIMATION_CONFIG,
        ...nextProps.animationConfig,
      };
      const { duration, delay, easing, isInteraction } = animationConfig;
      this.setState({ maskState: true }, () => {
        Animated.parallel([
          createAnimation({
            value: this.state.boxLeft,
            toValue: visible ? 0 : -this.range,
            duration,
            delay,
            easing,
            useNativeDriver: false,
            isInteraction,
          }),
          createAnimation({
            value: this.state.maskOpacity,
            toValue: +visible,
            duration,
            delay,
            easing,
            useNativeDriver: false,
            isInteraction,
          }),
        ]).start(() => {
          if (!visible) {
            this.setState({ maskState: false });
            this.endOnce = true;
          }
          typeof onStateChange === 'function' && onStateChange(visible);
        });
      });
      if (!visible) {
        this.boxLeft = 0;
      }
    }
  };

  range: number;
  boxLeft: number;
  endOnce: boolean;
  direction: string;

  _panResponder = PanResponder.create({
    onStartShouldSetPanResponder: () => true,
    onMoveShouldSetPanResponder: () => false,
    onPanResponderGrant: () => true,
    onPanResponderMove: () => false,
    onPanResponderRelease: _.throttle(evt => this._handleRelease(evt), 1000, {
      leading: true,
      trailing: false,
    }),
  });

  _handleRelease = (evt: GestureResponderEvent) => {
    const { visible, onMaskPress } = this.props;
    this.endOnce = false;
    if (this.endOnce || !visible || !evt || !evt.nativeEvent) return;
    typeof onMaskPress === 'function' && onMaskPress();
  };

  render() {
    const {
      maskStyle,
      drawerStyle,
      width,
      height,
      children,
      placement,
      hasMask,
      style,
    } = this.props;
    return (
      <View
        style={[
          {
            position: 'absolute',
            [['left', 'right'].includes(placement) ? 'top' : 'left']: 0,
            [placement]: 0,
          },
          style,
        ]}
      >
        {hasMask && this.state.maskState && (
          <Animated.View
            style={[
              { [['left', 'right'].includes(placement) ? 'top' : 'left']: 0 },
              maskStyle,
              { [placement]: 0, opacity: this.state.maskOpacity },
            ]}
            {...this._panResponder.panHandlers}
          />
        )}
        <Animated.View
          style={{
            flexDirection: 'row',
            position: 'absolute',
            [['left', 'right'].includes(placement) ? 'top' : 'left']: 0,
            [placement]: this.state.boxLeft,
          }}
        >
          <View style={[drawerStyle, { width, height }]}>{children}</View>
        </Animated.View>
      </View>
    );
  }
}
