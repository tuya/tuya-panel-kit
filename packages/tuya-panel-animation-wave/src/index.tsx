import React from 'react';
import { View, Animated } from 'react-native';
import Svg, { Path } from 'react-native-svg';
import { Utils } from 'tuya-panel-utils';
import { createAnimation, WAVE_DEFAULT_ANIMATION_CONFIG } from './utils';
import { WaveViewProps, WaveViewStateTypes } from './interface';

const AnimatedSvg = Animated.createAnimatedComponent(Svg);

const { convertX: cx, viewWidth } = Utils.RatioUtils;

/**
 * ---------+------------------------+
 * <-- P -->|<--    T    -->|        |______
 *          |   /\          |   /\   |  ^
 *          |  /  \         |  /  \  |  A
 *          | /    \        | /    \ |  |
 *          |/      \       |/      \|__V___
 *          |        \      /        |  ^
 *          |         \    /         |  |
 *          |          \  /          |  |
 *          |           \/           |  H
 *          |                        |  |
 *          |                        |  |
 * ---------+------------------------+__V___
 */

class Wave extends React.PureComponent<WaveViewProps, WaveViewStateTypes> {
  static defaultProps: WaveViewProps = {
    H: cx(50),
    waveParams: [
      { A: 25, T: 120, fill: '#64c3d6' },
      { A: 20, T: 100, fill: '#0092ba' },
    ],
    style: {},
    animated: true,
    animationConfig: WAVE_DEFAULT_ANIMATION_CONFIG,
  };

  constructor(props: WaveViewProps) {
    super(props);
    const { H, waveParams, animated, style } = this.props;
    /* eslint-disable prefer-object-spread */
    const wrapStyle = Object.assign(
      { width: cx(100), height: cx(100), overflow: 'hidden', backgroundColor: 'transparent' },
      style
    );

    this.state = {
      H: (wrapStyle.height * H) / 100,
      style: wrapStyle,
      waveParams,
    };

    this._animValues = [];
    this._animations = [];
    this._animated = animated || false;

    for (let i = 0; i < waveParams.length; i++) {
      this._animValues.push(new Animated.Value(0));
    }
  }

  componentDidMount() {
    this._animated && this.startAnim();
  }

  componentWillReceiveProps(nextProps: WaveViewProps) {
    if (this.props.animated !== nextProps.animated) {
      this._animated = nextProps.animated;
      this._animated && this.startAnim();
      !this._animated && this.stopAnim();
    }
  }

  componentWillUnmount() {
    this.stopAnim();
    this._animValues = null;
    this._animations = null;
  }

  _animValues: Animated.Value[];
  _animations: Animated.Value[];
  _animated: boolean;

  startAnim() {
    this.stopAnim();
    const animationConfig = {
      ...WAVE_DEFAULT_ANIMATION_CONFIG,
      ...this.props.animationConfig,
    };

    const { duration, delay, easing, useNativeDriver, isInteraction } = animationConfig;

    for (let i = 0; i < this._animValues.length; i++) {
      const anim = Animated.loop(
        createAnimation({
          value: this._animValues[i],
          toValue: 1,
          duration: duration + delay * i,
          delay: 0,
          easing,
          useNativeDriver,
          isInteraction,
        })
      );
      // @ts-ignore
      this._animations.push(anim);
      anim.start();
    }
    this._animated = true;
  }

  stopAnim() {
    for (let i = 0; i < this._animations.length; i++) {
      // @ts-ignore
      this._animations[i].stop();
      this._animations[i] = null;
    }
    this._animations = [];
    this._animated = false;
  }

  render() {
    const { H, waveParams, style } = this.state;
    const waves = [];
    const mergeStyle = { width: viewWidth, height: cx(120), ...style };
    for (let i = 0; i < waveParams.length; i++) {
      const { A, T, fill } = waveParams[i];
      const translateX = this._animValues[i].interpolate({
        inputRange: [0, 1],
        outputRange: [0, -2 * T],
      });
      // 保证水波足够长

      const scale = mergeStyle.width >= T ? Math.ceil(mergeStyle.width / T) : 1;
      let d = `M 0 ${A} Q ${T / 4} ${2 * A} `;
      for (let j = 1; j < 6 * scale + 1; j++) {
        d += `${(T * j) / 2} ${A} ${j === 6 * scale ? 'V' : 'T'} `;
      }
      d += ` ${A + H} H 0 Z`;
      const wave = (
        <AnimatedSvg
          key={i}
          style={{
            width: 3 * scale * T,
            height: A + H,
            position: 'absolute',
            left: 0,
            bottom: 0,
            transform: [{ translateX }],
          }}
          preserveAspectRatio="xMinYMin meet"
          viewBox={`0 0 ${3 * scale * T} ${A + H}`}
        >
          <Path d={d} fill={fill} />
        </AnimatedSvg>
      );
      waves.push(wave);
    }

    return <View style={mergeStyle}>{waves}</View>;
  }
}

export default Wave;
