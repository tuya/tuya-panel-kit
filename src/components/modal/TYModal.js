/* eslint-disable react/no-array-index-key */
import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components/native';
import {
  Modal,
  TouchableOpacity,
  Animated,
  StyleSheet,
  ViewPropTypes,
  View,
  Platform,
  KeyboardAvoidingView,
} from 'react-native';
import { CoreUtils, RatioUtils } from '../../utils';

const { get } = CoreUtils;

const ALIGN = {
  top: 'flex-start',
  center: 'center',
  bottom: 'flex-end',
};

const { isIphoneX } = RatioUtils;

const ios = Platform.OS === 'ios';

/* istanbul ignore next */
const StyledMask = styled(TouchableOpacity)`
  background-color: ${props => get(props, 'theme.global.mask', 'rgba(0, 0, 0, 0.7)')};
`;

class TYModal extends React.Component {
  static propTypes = {
    /**
     * modal出现的动画效果
     */
    animationType: PropTypes.oneOf(['fade', 'none']),
    /**
     * modal默认出现的位置
     */
    alignContainer: PropTypes.oneOf(['top', 'center', 'bottom']),
    /**
     * 点击遮罩回调
     */
    onMaskPress: PropTypes.func,
    /**
     * modal弹出内容的样式
     */
    modalChildStyle: ViewPropTypes.style,
    /**
     * 遮罩层样式
     */
    maskStyle: ViewPropTypes.style,
    /**
     * 是否有遮罩层
     */
    mask: PropTypes.bool,
    /**
     * 是否只显示最后一个弹出的 modal
     */
    onlyLastModalVisible: PropTypes.bool,
    /**
     * 是否弹出键盘自适应
     */
    useKeyboardView: PropTypes.bool,

    /**
     * Modal 组件销毁回调事件，一般用于在弹窗销毁后跳转新的 native 页面
     * @platform iOS
     */
    onDismiss: PropTypes.func,
  };

  static defaultProps = {
    animationType: 'fade',
    alignContainer: 'bottom',
    onMaskPress: () => {},
    maskStyle: null,
    mask: true,
    modalChildStyle: null,
    onlyLastModalVisible: true,
    useKeyboardView: false,
  };

  constructor(props) {
    super(props);
    this._childRef = {};
    this.state = {
      fade: new Animated.Value(0),
    };
  }

  componentDidMount() {
    Animated.timing(this.state.fade, {
      duration: 200,
      toValue: 1,
    }).start();
  }

  _handleMaskPress = () => {
    const { onMaskPress, children } = this.props;
    const childMotionType = get(this, `_childRef.${children.length - 1}.props.motionType`, 'none');
    const childMaskPress = get(this, `_childRef.${children.length - 1}._handleMaskPress`);
    if (childMotionType !== 'none' && typeof childMaskPress === 'function') {
      /* istanbul ignore next */
      childMaskPress();
    } else {
      typeof onMaskPress === 'function' && onMaskPress();
    }
  };

  renderNoMaskModal = () => {
    const { onlyLastModalVisible, activeIdx, children, animationType, alignContainer } = this.props;
    const maskContainer = [
      styles.maskContainer,
      alignContainer && { justifyContent: ALIGN[alignContainer] },
      { opacity: animationType === 'fade' ? this.state.fade : 1 },
    ];
    return children.map((child, idx) => {
      const childStyle = [
        maskContainer,
        onlyLastModalVisible && { display: idx === activeIdx ? 'flex' : 'none' },
      ];
      return (
        <Animated.View key={idx} style={childStyle} pointerEvents="box-none">
          {child}
        </Animated.View>
      );
    });
  };

  renderMaskModal = () => {
    const {
      visible,
      onDismiss,
      onlyLastModalVisible,
      activeIdx,
      children,
      animationType,
      onMaskPress,
      alignContainer,
      maskStyle,
      modalChildStyle,
      useKeyboardView,
      ...props
    } = this.props;
    const maskContainer = [
      styles.maskContainer,
      alignContainer && { justifyContent: ALIGN[alignContainer] },
      maskStyle,
    ];
    return (
      <Modal
        animationType={animationType}
        onRequestClose={this._handleMaskPress}
        visible={onlyLastModalVisible ? visible : true}
        onDismiss={onDismiss}
        {...props}
        transparent={true}
      >
        {children.map((child, idx) => {
          const childStyle = [
            maskContainer,
            onlyLastModalVisible && { display: idx === activeIdx ? 'flex' : 'none' },
          ];
          let modalChild = child;
          if (!Array.isArray(child)) {
            modalChild = React.cloneElement(child, {
              onMaskPress: get(child, 'props.onMaskPress', onMaskPress),
              ref: /* istanbul ignore next */ ref => {
                this._childRef[idx] = ref;
              },
            });
          }
          return (
            <StyledMask
              key={idx}
              style={childStyle}
              onPress={this._handleMaskPress}
              activeOpacity={1}
            >
              {ios && useKeyboardView ? (
                <KeyboardAvoidingView
                  behavior="padding"
                  style={{
                    flex: 1,
                    alignItems: 'center',
                    justifyContent: 'center',
                    paddingTop: isIphoneX ? 20 : 0,
                  }}
                >
                  <View onStartShouldSetResponder={() => true} style={modalChildStyle}>
                    {modalChild}
                  </View>
                </KeyboardAvoidingView>
              ) : (
                <View onStartShouldSetResponder={() => true} style={modalChildStyle}>
                  {modalChild}
                </View>
              )}
            </StyledMask>
          );
        })}
      </Modal>
    );
  };

  render() {
    const { mask } = this.props;
    if (!mask) {
      return this.renderNoMaskModal();
    }
    return this.renderMaskModal();
  }
}

const styles = StyleSheet.create({
  maskContainer: {
    position: 'absolute',
    top: 0,
    bottom: 0,
    left: 0,
    right: 0,
  },
});

export default TYModal;
