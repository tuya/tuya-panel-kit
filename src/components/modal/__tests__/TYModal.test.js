import React from 'react';
import { View, Modal, Platform } from 'react-native';
import { shallow, mount } from 'enzyme';
import 'react-native/Libraries/Animated/src/bezier';
import TYModal from '../TYModal';

function setup() {
  const props = {
    mask: true,
    useKeyboardView: false,
  };
  const wrapper = shallow(
    <TYModal {...props}>
      <View />
      <View />
    </TYModal>
  );
  const instance = wrapper.instance();
  return { wrapper, instance };
}

describe('TYModal Component', () => {
  it('renders correctly', () => {
    Platform.OS = 'ios';
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
    const secondView = wrapper
      .findWhere(c => c.name() === 'View' && !!c.prop('onStartShouldSetResponder') === true)
      .last();
    secondView.simulate('startShouldSetResponder');
    wrapper.setProps({ mask: false });
    expect(wrapper).toMatchSnapshot();
  });
  it('renders correctly with useKeyboardView', () => {
    const wrapper = shallow(
      <TYModal mask={true} useKeyboardView={true}>
        <View />
        <View />
      </TYModal>
    );
    expect(wrapper).toMatchSnapshot();
    const firstView = wrapper
      .findWhere(c => c.name() === 'View' && !!c.prop('onStartShouldSetResponder') === true)
      .first();
    firstView.simulate('startShouldSetResponder');
  });
  it('call _handleMaskPress method', () => {
    const { wrapper } = setup();
    const modal = wrapper.find(Modal);
    modal.simulate('requestClose');
  });
});
