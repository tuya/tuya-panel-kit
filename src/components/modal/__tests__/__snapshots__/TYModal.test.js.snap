// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TYModal Component renders correctly 1`] = `
<Component
  animationType="fade"
  hardwareAccelerated={false}
  mask={true}
  onRequestClose={[Function]}
  transparent={true}
  visible={true}
>
  <Styled(TouchableOpacity)
    activeOpacity={1}
    key="0"
    onPress={[Function]}
    style={
      Array [
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "justifyContent": "flex-end",
          },
          null,
        ],
        Object {
          "display": "none",
        },
      ]
    }
  >
    <View
      onStartShouldSetResponder={[Function]}
      style={null}
    >
      <View
        onMaskPress={[Function]}
      />
    </View>
  </Styled(TouchableOpacity)>
  <Styled(TouchableOpacity)
    activeOpacity={1}
    key="1"
    onPress={[Function]}
    style={
      Array [
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "justifyContent": "flex-end",
          },
          null,
        ],
        Object {
          "display": "none",
        },
      ]
    }
  >
    <View
      onStartShouldSetResponder={[Function]}
      style={null}
    >
      <View
        onMaskPress={[Function]}
      />
    </View>
  </Styled(TouchableOpacity)>
</Component>
`;

exports[`TYModal Component renders correctly 2`] = `
Array [
  <AnimatedComponent
    key="0"
    pointerEvents="box-none"
    style={
      Array [
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "justifyContent": "flex-end",
          },
          Object {
            "opacity": 0,
          },
        ],
        Object {
          "display": "none",
        },
      ]
    }
  >
    <View />
  </AnimatedComponent>,
  <AnimatedComponent
    key="1"
    pointerEvents="box-none"
    style={
      Array [
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "justifyContent": "flex-end",
          },
          Object {
            "opacity": 0,
          },
        ],
        Object {
          "display": "none",
        },
      ]
    }
  >
    <View />
  </AnimatedComponent>,
]
`;

exports[`TYModal Component renders correctly with useKeyboardView 1`] = `
<Component
  animationType="fade"
  hardwareAccelerated={false}
  mask={true}
  onRequestClose={[Function]}
  transparent={true}
  visible={true}
>
  <Styled(TouchableOpacity)
    activeOpacity={1}
    key="0"
    onPress={[Function]}
    style={
      Array [
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "justifyContent": "flex-end",
          },
          null,
        ],
        Object {
          "display": "none",
        },
      ]
    }
  >
    <KeyboardAvoidingView
      behavior="padding"
      keyboardVerticalOffset={0}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
          "paddingTop": 20,
        }
      }
    >
      <View
        onStartShouldSetResponder={[Function]}
        style={null}
      >
        <View
          onMaskPress={[Function]}
        />
      </View>
    </KeyboardAvoidingView>
  </Styled(TouchableOpacity)>
  <Styled(TouchableOpacity)
    activeOpacity={1}
    key="1"
    onPress={[Function]}
    style={
      Array [
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "justifyContent": "flex-end",
          },
          null,
        ],
        Object {
          "display": "none",
        },
      ]
    }
  >
    <KeyboardAvoidingView
      behavior="padding"
      keyboardVerticalOffset={0}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
          "paddingTop": 20,
        }
      }
    >
      <View
        onStartShouldSetResponder={[Function]}
        style={null}
      >
        <View
          onMaskPress={[Function]}
        />
      </View>
    </KeyboardAvoidingView>
  </Styled(TouchableOpacity)>
</Component>
`;
